packages/shared/blockchain/evm-based/src/lib/evm-based.utils.ts

```diff
@@ -86,6 +89,30 @@ export const evmBasedUtils = {
     return signedTransaction.rawTransaction
   },
 
+  validateSenderBalance: async (client: Web3, privateKey: string, tx: TransactionConfig) => {
+    const { gas, gasPrice, nonce, ...txWithoutGas } = tx
+    let estimate: BigNumber
+    try {
+      estimate = new BigNumber(await client.eth.estimateGas(txWithoutGas))
+    } catch (e) {
+      if (gas) estimate = new BigNumber(gas as string)
+      else throw new EvmBasedSdkError({ error: e as Error })
```

Comment:
you must throw more specific error with error message, so inside core api we willl be able to detect itt properly. something like cannot estimate gas, reason:..., e